src/utils/logging.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Logging is a helper class to manage log files; they can be viewed in the
+ * Troubleshooting tab in the UI.
+ *
+ * Usage:
+ *
+ * import Logging from '.../logging';
+ *
+ * const topicLog = Logging('topic');
+ * await Logging.topic('Log string goes here');
+ * // Equivalent to `await topicLog.log('Log string goes here');
+ * // The class has a `path` member:
+ * fs.readFile(Logging.topic.path, ...);
+ */
+
+import fs from 'fs';
+import path from 'path';
+import stream from 'stream';
+
+import XDGAppPaths from 'xdg-app-paths';
+const paths = XDGAppPaths({ name: 'rancher-desktop' });
+const logDir = path.join(paths.runtime() || paths.data(), 'logs');
```

Comment:
`paths.runtime()` is undefined on Windows, so it will fall back to `paths.data()`, which will be on the roaming profile. On a computer connected to a domain this data is copied back to the domain controller on logout, so it can migrate to a different machine with the user.

I would suggest to use `paths.state()` instead because it is stored in the local profile.